src/models/json-api.model.ts

```diff
@@ -65,22 +66,35 @@ export class JsonApiModel {
     return Reflect.getMetadata('JsonApiModelConfig', this.constructor);
   }
 
+
   private parseHasMany(data: any, included: any, level: number): void {
     let hasMany: any = Reflect.getMetadata('HasMany', this);
     if (hasMany) {
       for (let metadata of hasMany) {
         let relationship: any = data.relationships ? data.relationships[metadata.relationship] : null;
         if (relationship && relationship.data && relationship.data.length > 0) {
-          let typeName: string = relationship.data[0].type;
-          let modelType: ModelType<this> = Reflect.getMetadata('JsonApiDatastoreConfig', this._datastore.constructor).models[typeName];
-          if (modelType) {
-            let relationshipModel: JsonApiModel[] = this.getHasManyRelationship(modelType, relationship.data, included, typeName, level);
-            if (relationshipModel.length > 0) {
-              this[metadata.propertyName] = relationshipModel;
+          let allModels: any = [];
+          let modelTypesFetched: any = [];
+          for (let typeIndex of Object.keys(relationship.data)) {
+            let typeName: string = relationship.data[typeIndex].type;
+            if (!includes(modelTypesFetched, typeName)) {
+                modelTypesFetched.push(typeName);
+                let modelType: ModelType<this> = Reflect.getMetadata('JsonApiDatastoreConfig', this._datastore.constructor).models[typeName];
+                if (modelType) {
+                  let relationshipModel: JsonApiModel[] = this.getHasManyRelationship(modelType, relationship.data, included, typeName, level);
+                  if (relationshipModel.length > 0) {
```

Comment:
could you define `allModels` als var and write this as `allModels = allModels.concat(relationshipModels)`?

Reply:
Rename to `relationshipModels`.